source/common/stats/timespan_impl.cc

```diff
@@ -0,0 +1,47 @@
+#include "common/stats/timespan_impl.h"
+
+namespace Envoy {
+namespace Stats {
+
+HistogramCompletableTimespanImpl::HistogramCompletableTimespanImpl(Histogram& histogram,
+                                                                   TimeSource& time_source)
+    : time_source_(time_source), histogram_(histogram), start_(time_source.monotonicTime()) {
+  if (!histogram.active()) {
+    return;
+  }
+
+  switch (histogram.unit()) {
+  case Histogram::Unit::Unspecified:
+  case Histogram::Unit::Bytes:
+    ASSERT(0);
+  case Histogram::Unit::Microseconds:
+  case Histogram::Unit::Milliseconds:
+    return;
+  }
+
+  ASSERT(0);
```

Comment:
this should be NOT_REACHED if this is required for compilation.

Reply:
My bad, did not notice this macro even though it was right there next to assert. Fixed throughout.